File: app/src/test/java/de/jodamob/kotlin/droidcon/b/ImportantClassCheckingSomethingTest.java
package de.jodamob.kotlin.droidcon.b;

import org.junit.Test;


public class ImportantClassCheckingSomethingTest {

    @Test(expected = IllegalArgumentException.class)
    public void should_throw_if_empty() {
        ImportantClassCheckingSomething tested = new ImportantClassCheckingSomething();
        tested.dontletItHappen("");
    }
}

File: app/src/test/java/de/jodamob/kotlin/droidcon/b/ImportantClassCheckingSomethingTest.kt
package de.jodamob.kotlin.droidcon.b

import org.amshove.kluent.`with message`
import org.amshove.kluent.shouldThrow
import org.junit.Test
import kotlin.test.assertFailsWith


class ImportantClassCheckingSomethingTest {

    @Test
    fun `should throw if empty normal version`() {
        val tested = ImportantClassCheckingSomething()
        
        assertFailsWith(IllegalArgumentException::class) {
            tested.dontletItHappen("")
        }
    }

    @Test
    fun `should throw if empty kluent version`() {
        val tested = ImportantClassCheckingSomething()
        val function = {tested.dontletItHappen("")}

        function shouldThrow IllegalArgumentException::class `with message` "was empty"
    }
}